sql/system_variables.go

```diff
@@ -0,0 +1,2459 @@
+// Copyright 2021 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sql
+
+import "math"
+
+// SystemVariableScope represents the scope of a system variable.
+type SystemVariableScope byte
+const (
+	// SystemVariableScope_Global is set when the system variable exists only in the global context.
+	SystemVariableScope_Global SystemVariableScope = iota
+	// SystemVariableScope_Session is set when the system variable exists only in the session context.
+	SystemVariableScope_Session
+	// SystemVariableScope_Both is set when the system variable exists in both the global and session contexts.
+	SystemVariableScope_Both
+)
+
+// SystemVariable represents a system variable.
+type SystemVariable struct {
+	// Name is the name of the system variable.
+	Name string
+	// Scope defines the scope of the system variable, which is either Global, Session, or Both.
+	Scope SystemVariableScope
+	// Dynamic defines whether the variable may be set during runtime.
+	Dynamic bool
```

Comment:
Prefer Writable

Reply:
So I'm actually mimicking the terminology used here:
https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html
I can rework the comment, but I'd prefer to keep the terminology the same. Dynamic is used in a lot of places in the documentation.
https://dev.mysql.com/doc/refman/8.0/en/dynamic-system-variables.html